probe-rs/src/probe/stlink/mod.rs

```diff
@@ -171,7 +171,8 @@ impl DebugProbe for STLink {
             TIMEOUT,
         )?;
 
-        Self::check_status(&buf)
+        Self::check_status(&buf)?;
```

Comment:
Is there a reason for this change I do not see?

Reply:
Ah, found it. I did this change too in my hard break fix, but reverted it. I am not sure why anymore ... maybe we'll see later on if there was another issue. Until then this is ok to me.